data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/OpenSearchPluginAttributesMapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import org.opensearch.dataprepper.logstash.exception.LogstashConfigurationException;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class OpenSearchPluginAttributesMapper extends AbstractLogstashPluginAttributesMapper {
+    protected static final String LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME = "index";
+    protected static final String REGEX_EXTRACTING_DATETIME_PATTERN = "%\\{(.*?)\\}";
+
+    @Override
+    protected void mapCustomAttributes(List<LogstashAttribute> logstashAttributes, LogstashAttributesMappings logstashAttributesMappings, Map<String, Object> pluginSettings) {
+
+        final LogstashAttribute logstashIndexAttribute = logstashAttributes.stream()
+                .filter(a -> a.getAttributeName().equals(LOGSTASH_OPENSEARCH_INDEX_ATTRIBUTE_NAME))
+                .findFirst()
+                .orElseThrow(() -> new LogstashConfigurationException("Index attribute not found"));
```

Comment:
I see, looks like I did not update this in the new revision, will do now. Thanks @cmanning09 @dlvenable

Reply:
@sshivanii, the current approach is not handling the default values for the [elasticsearch output plugin](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index) or the opensearch output plugin. We need to support a migration of these indices.

Regarding the opensearch output plugin: we need to dive deep here to understand what the default values our. The documentation is not clear at this time.

Reply:
@cmanning09 Regarding the default values for index in [Logstash ES output plugin](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index), it's dependent on the `ecs_compatibility` attribute which we do not support in our sink plugin so it will not be added to the converted `.yaml` file. Since we don't support ecs compatibility, I believe these default values should not be added.

I dug deeper into the OpenSearch code to find they only have a default index depending on `ecs` being enabled/disabled. Otherwise they throw an exception `Unsupported ECS compatibility`
https://github.com/opensearch-project/logstash-output-opensearch/blob/main/lib/logstash/outputs/opensearch.rb#L417

Reply:
I'd like to suggest that when the index name is not supplied, Data Prepper uses the index name for when ECS is disabled. ECS compatibility is a broader topic than this PR is covering, and I don't think it should hinder this PR.

Thus, if the `index` is not present, the converter can just set the index to `logstash-%{uuuu.MM.dd}`. The current two tests which have no index value would both need to change to verify this new behavior.

Reply:
Also, please verify that this is also the default of the `amazon_es` plugin.

Reply:
I think the best solution here is to include a `defaultAttributes` property to the Logstash configuration mapping YAML. Other plugins may need to have default values where the Logstash default is different from the Data Prepper default.

As this problem exists in the current converter and this change introduces no regressions, I'd rather we wrap up this PR and support this more generically.

Reply:
Checked the default index attribute value in `amazon_es` plugin and it is `logstash-%{+YYYY.MM.dd}` so the converted value will be `logstash-%{yyyy.MM.dd}`. This is different from the `elasticsearch` output plugin default, and the plugin name is not known at this time.
Reference: `https://github.com/awslabs/logstash-output-amazon_es/blob/master/lib/logstash/outputs/amazon_es/common_configs.rb#L17`